luigi_pipeline/lib/model/seqr_mt_schema.py

```diff
@@ -109,31 +109,48 @@ def genotypes(self):
     def samples_no_call(self):
         return self._genotype_filter_samples(lambda g: g.num_alt == -1)
 
-    @row_annotation(fn_require=genotypes, multi_annotation=True)
+    @row_annotation(fn_require=genotypes)
     def samples_num_alt(self, start=1, end=3, step=1):
-        # NOTE: Multiple annotations.
-        return {
-            'samples_num_alt_%i' % i: self._genotype_filter_samples(lambda g: g.num_alt == i)
+        return hl.struct(**{
+            '%i' % i: self._genotype_filter_samples(lambda g: g.num_alt == i)
             for i in range(start, end, step)
-        }
+        })
 
-    @row_annotation(fn_require=genotypes, multi_annotation=True)
+    @row_annotation(fn_require=genotypes)
     def samples_gq(self, start=0, end=95, step=5):
-        # NOTE: Multiple annotations.
-        return {
-            'samples_gq_%i_to_%i' % (i, i+step): self._genotype_filter_samples(lambda g: ((g.gq >= i) & (g.gq < i+step)))
+        # struct of x_to_y to a set of samples in range of x and y for gq.
+        return hl.struct(**{
+            '%i_to_%i' % (i, i+step): self._genotype_filter_samples(lambda g: ((g.gq >= i) & (g.gq < i+step)))
             for i in range(start, end, step)
-        }
+        })
 
-    @row_annotation(fn_require=genotypes, multi_annotation=True)
+    @row_annotation(fn_require=genotypes)
     def samples_ab(self, start=0, end=45, step=5):
-        # NOTE: Multiple annotations.
-        return {
-            'samples_ab_%i_to_%i' % (i, i+step): self._genotype_filter_samples(
+        # struct of x_to_y to a set of samples in range of x and y for ab.
+        return hl.struct(**{
+            '%i_to_%i' % (i, i+step): self._genotype_filter_samples(
                 lambda g: ((g.num_alt == 1) & ((g.ab*100) >= i) & ((g.ab*100) < i+step))
             )
             for i in range(start, end, step)
-        }
+        })
+
+    def elasticsearch_row(self):
+        """
+        Prepares the mt to export using ElasticsearchClient V02.
+        - Flattens nested structs
+        - drops locus and alleles key
+
+        TODO:
+        - Call validate
+        - when all annotations are here, whitelist fields to send instead of blacklisting.
+        :return:
+        """
+        # Converts nested structs into one field, e.g. {a: {b: 1}} => a.b: 1
+        table = self.mt.rows().flatten()
+        # When flattening, the table is unkeyed, which causes problems because our locus and alleles should not
+        # be normal fields. We can also re-key, but I believe this is computational?
+        table = table.drop(table.locus, table.alleles)
+        return table
```

Comment:
@hanars, @bw2 This is the simplified code. It uses `flatten` so we don't have to manually recurse through, and by using the v02 elasticsearch client I believe we have all of the ES logic (except for the temp loading node switch).

Reply:
looks fine to me, but @bw2 should probably also take a look